sycl/include/CL/sycl/item.hpp

```diff
@@ -30,6 +30,16 @@ template <int dimensions> class range;
 ///
 /// \ingroup sycl_api
 template <int dimensions = 1, bool with_offset = true> class item {
+#ifndef __SYCL_DISABLE_ITEM_TO_INT_CONV__
+  /* Helper class for conversion operator. Void type is not suitable. User
+   * cannot even try to get address of the operator __private_class(). User
+   * may try to get an address of operator void() and will get the
+   * compile-time error */
+  class __private_class;
+
+  template <bool B, typename T>
+  using EnableIfT = detail::conditional_t<B, T, __private_class>;
```

Comment:
Because static_assert doesn't disable the operator. We need that `operator size_t` participates in overload resolution only if dimensions == 1. In other cases it may be solved by `enable_if` but that case is different. To use `enable_if` we need to templatize the `operator size_t` itself. But templatized conversion operators have other limitations that don't fit the expected behavior. We previously struggled with the same issue for `id` class